File: synapse-transport/src/main/java/com/synapse/transport/feature/experiment/dto/response/ExperimentDatasetResponse.java
package com.synapse.transport.feature.experiment.dto.response;

import java.util.UUID;

import com.synapse.transport.common.enums.ConfigStatus;
import com.synapse.transport.common.enums.DatasetProvider;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ExperimentDatasetResponse {
    private UUID experimentId;

    private ConfigStatus datasetStatus;
    private DatasetProvider provider;
    private String datasetName;

    private Integer trainSampleCount;
    private Integer testSampleCount;

    private String inputShape;
    private String outputShape;

    private Integer batchSize;
    private Integer numWorkers;
    private Boolean shuffle;
    private Boolean pinMemory;
    private Boolean dropLast;
    private Integer prefetchFactor;
    private Boolean persistentWorkers;

    private String lastValidationError;
}

File: synapse-ui/src/types/api/experiment/experiment-dataset.response.ts
import type { ConfigStatus, DatasetProvider } from "../../enums/transport.enums";

export interface ExperimentDatasetResponse {
    experimentId: string;
    datasetStatus: ConfigStatus;
    provider: DatasetProvider | null;
    datasetName: string | null;
    trainSampleCount: number | null;
    testSampleCount: number | null;
    inputShape: string | null;
    outputShape: string | null;
    batchSize: number | null;
    numWorkers: number | null;
    shuffle: boolean | null;
    pinMemory: boolean | null;
    dropLast: boolean | null;
    prefetchFactor: number | null;
    persistentWorkers: boolean | null;
    lastValidationError: string | null;
}
